lib/spack/spack/test/architecture.py

```diff
@@ -159,3 +160,23 @@ def test_user_input_combination(config):
         )
     res = all(results)
     assert res
+
+
+def test_swapping_to_frontend_environment(config, mock_configure, monkeypatch):
+
+    spec = Spec("mpileaks")
+    spec.concretize()
+
+    def mock_load_module(mod):
+        os.environ["MODULE_LOADED"] = "frontend"
```

Comment:
Requests: 

* set `os.environ["MODULE_LOADED"] = mod`
* then the below `assert` can check that the module associated with the spec architecture is loaded
* if you update the generated test mock configure script to return `MODULE_LOADED`, then I think it would be useful to add an assert that the value of `MODULE_LOADED` during `configure` was set to the frontend arch module